hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/storage/row/HoodieRowCreateHandle.java

```diff
@@ -184,7 +185,7 @@ private void writeRow(InternalRow row) {
       UTF8String writeCommitTime = shouldPreserveHoodieMetadata ? row.getUTF8String(HoodieRecord.COMMIT_TIME_METADATA_FIELD_ORD)
           : commitTime;
 
-      InternalRow updatedRow = new HoodieInternalRow(writeCommitTime, seqId, recordKey,
+      InternalRow updatedRow = SparkAdapterSupport$.MODULE$.sparkAdapter().createInternalRow(writeCommitTime, seqId, recordKey,
```

Comment:
Have you checked if the performance is the same after replacing with `HoodieInternalRow::new` with `SparkAdapterSupport$.MODULE$.sparkAdapter().createInternalRow`.